### ISSUE TITLE
headerCheckboxable option isn't working as expected.

### ISSUE THREAD

**Issue by rafaelss95** [swimlane/ngx-datatable#724]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**
The `headerCheckboxable` option isn't working as expected.

**Expected behavior**
The header checkbox option must "watch" the others checkboxes, as soon as all checkboxes are ticked it must be checked also, otherwise, if there's one checkbox deselected it should be automatically deselected also.

**Reproduction of the problem**
Go to [#checkbox demo](http://swimlane.github.io/ngx-datatable/#chkbox-selection) and follow the steps:

1. Select the header checkbox;
2. Deselect a register (now the bug starts, you'll note that the header doesn't deselects automatically, but it would since there's one register deselected);
3. Deselect the header checkbox (It will check all checkboxes and only he will be deselected);
4. Select the header checkbox (It will uncheck all checkboxes and only he will be selected);

* **Table version:** 9.0.0

* **Angular version:** 4.0.03

**Comment by nyankov** [comment]
I observed the same behavior.

Same issue was reported by me here: https://github.com/swimlane/ngx-datatable/issues/659 . It's marked as solved, but I just tried with version 9.0.0 and it's still actual.

**Comment by luistak** [comment]
Same issue with my project.

**Comment by mvhecke** [comment]
This is a duplicate of #659. I've created a pull request (#719) to fix this issue.

**Comment by amcdnl** [comment]
I added some comments to the review, if we can get together on those items happy to merge!

**Comment by rafaelss95** [comment]
Working for me in the latest (9.1.0).

**Comment by MorlaRamakrishna** [comment]
I am working with Angular 2, can you please help me in implementing this with Datatable@6.3.0 and I am able to succeed in presenting Show/Hide columns, sorting, pagination, Individual selection with the check box. But I am failing to get **select all**.

**Comment by mvhecke** [comment]
@MorlaRamakrishna It's fixed since version 9.1.0 but that requires you to switch to Angular 4. You can check what has changed in #719.

**Comment by MorlaRamakrishna** [comment]
Is there any solution to fix this issue without upgrading to Angular 4.
